### ISSUE TITLE
[Death Knight] Blood Presence - Thorns, Immolation and several itemspells

### ISSUE THREAD

**Issue by Darthblaider** [dalaranwow/dalaran-wow#3829]
The healing effect of Blood Presence and Improved Blood Presence for Death Knights is not working to the fullest.

The spell text itself as wall as any source I can find make it seem as though it's supposed to heal you for ALL damage dealt, but on Algalon, this surely isn't true. 

The largest issue is that Death and Decay doesn't trigger any healing AT ALL from blood presence, talented or not. This is hugely important for any death knight, especially in AoE fights in raids, and even more so important in pvp scenarios.

The idea of blood presence healing from ALL damage is definitely not present in many things. For instance, it will work with engineering rockets fired, but not with damage dealt by the hand of ragnaros's thorns-like affect(when scaled up with spellpower).  At the same time though, the effect gains the 15% damage bonus from the presence. On the same note, thorns from a druid does not heal either, and does NOT benefit from the damage bonus. This shows you that the Hand of Rag's effect isn't the same as normal thorns. Unlike that thorns buff, the "Immolate" from the hammer is being counted as the players damage, and not that of a buff's. I believe that if this is the case, anything that benefits from the damage bonus of blood presence, should also benefit in the healing bonus as well.

This brought me to researching what is and isn't supposed to heal you when doing damage. the ONLY sources I could find are that of the actual spell text which reads "Strengthens the Death Knight with the presence of blood, increasing damage by 15% and healing the Death Knight by 4% of damage dealt. Only one Presence may be active at a time."

Steps to reproduce!
1)have a death knight
2)punch stuff>get healed
3)death and decay stuff>don't get healed
4)let stuff punch you and die to your thorns from a druid>don't get healed
5)Get some spellpower and let stuff punch you with hand of rag>don't get healed.

In conclusion, the healing affects need to be fixed, and any damage that benefits from the 15% bonus should, in my opinion, also be healing the player.

*this was tested with and without the Improved Blood Presence talent*

**Comment by Darthblaider** [comment]
*this was tested with and without the Improved Blood Presence talent*

**Comment by Rushor** [comment]
add the spellids for talents/spells from http://wotlk.openwow.com

**Comment by Darthblaider** [comment]
Blood Presence: http://wotlk.openwow.com/spell=48266
Thorns rank I used: http://wotlk.openwow.com/spell=26992
Death and Decay: http://wotlk.openwow.com/spell=49938
Immolation from Hand of Rag: http://wotlk.openwow.com/spell=21142#used-by-item

**Comment by robinsch** [comment]
Can you find a video from retail where we can see that for example DnD should trigger the healing?

**Comment by Darthblaider** [comment]
https://www.youtube.com/watch?v=NEFYkeHo_7k at about 1:20

**Comment by robinsch** [comment]
On PTR.

**Comment by Jokey11** [comment]
Death and Decay was fixed and it shall be applied to the live realm soon enough.
However, Thorns and Immolation from Hand of Ragnaros are not fixed yet, so I shall re-open this report.

**Comment by Darthblaider** [comment]
There are other similar items that have affects like the hand of rag

cloak of flames: http://wotlk.openwow.com/item=3475
Naglering: http://wotlk.openwow.com/item=11669
Razor Gauntlets: http://wotlk.openwow.com/item=18326
Drillborer Disc: http://wotlk.openwow.com/item=17066

**Comment by Edzero** [comment]
Marking this as Waiting for Changelog without the items mentioned in Blaider's last post so that the main fix for DnD can be on live realm sooner.

**Comment by TeridanDW** [comment]
Hello after some tests on ptr the fix works as intended.

Note: Proof for the missing healing was provided only for death and decay spell (it heals now) the rest of the items/spells are not suppose to scale with sp and heal (to my knowledge) and until proof is provided for oposite they won't heal/scale.
